### DISCUSSION TITLE
Add a new application Error

### DISCUSSION THREAD

**Discussion by sachinbhartiaphro** [codemagic-ci-cd/codemagic-docs#2540] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

**we are using this api** 
curl -H "Content-Type: application/json" \
     -H "x-auth-token: <API Token>" \
     -d '{
       "repositoryUrl": "git@github.com:my-organization/my-repo.git"
     }' \
     -X POST https://api.codemagic.io/apps  

**Docs url :** https://docs.codemagic.io/rest-api/applications/

**Error Screenshot** 
<img width="677" alt="Screenshot 2024-05-18 at 6 26 57 PM" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/146416380/e9bf3d8d-892b-4df8-a9c5-d175c42d3a18">

### Expected results

Successful completion of cloning, iOS build creation, and automatic upload to TestFlight.

### Actual results

Cannot access the repository https://bitbucket.org/sachin_aphroecs/aphromobiletest.git
Failed to fetch tags and branches with 'git ls-remote': remote: Invalid credentials fatal: Authentication failed for 'https://bitbucket.org/sachin_aphroecs/aphromobiletest.git/'

**Comment by DeepakAphro121** [comment]
i am getting same issue

**Comment by dtrdic** [comment]
hey @sachinbhartiaphro if your repositories are within an internal network / behind a firewall, the following IP addresses must be allowed through the firewall (whitelisted)
```
66.185.17.80/29 
199.7.162.128/29 
207.254.42.240/29 
34.74.234.56/32 
35.185.76.207/32
```

**Reply by sachinbhartiaphro** [reply]
Hi,

We have purchased a Premium plan on Bitbucket and have added the whitelisted IP. Additionally, we have bought a paid Teams plan on Codemagic. However, we are still facing issues with permissions on Bitbucket, which is preventing us from proceeding.

Could you please assist us in resolving this issue as soon as possible?

**Reply by dtrdic** [reply]
Hi @sachinbhartiaphro, since your repository is private you need to provide `sshKey` as well. So your API would look like this;
```
 curl -H "Content-Type: application/json" \
       -H "x-auth-token: <API Token>" \
       -d '{
         "repositoryUrl": "git@github.com:my-organization/my-repo.git",
         "sshKey": {
           "data": "St89hgb-BASE64-ENCODED-SSH-KEY-FILE-H4ga7jgf==", 
           "passphrase": null
         }
       }' \
       -X POST https://api.codemagic.io/apps 
```
Please [check this](https://docs.codemagic.io/rest-api/applications/#add-a-new-application-from-a-private-repository) for reference

**Comment by sachinbhartiaphro** [comment]
We are following the same as given in the docs.  Please check my curl.
Docs url :       https://docs.codemagic.io/rest-api/applications/#add-a-new-application-from-a-private-repository
curl :        curl --location 'https://api.codemagic.io/apps' \
--header 'Content-Type: application/json' \
--header 'x-auth-token: XXX' \
--data '{
         "repositoryUrl": "Repo-url",
           "projectType": "flutter-app",
         "sshKey": {
           "data": "SHA-key",
           "passphrase": null
            
 }
       }'

**Reply by mikhail-tokarev** [reply]
please be careful when you share commands you use. I've removed your token from the message and will revoke it so you need to use a new one.

**Reply by mikhail-tokarev** [reply]
@sachinbhartiaphro additionally to my previous message I highly recommend you to rotate your SSH keys. 

re the error: looks like you provided a public key instead of private one

**Comment by sachinbhartiaphro** [comment]
ok sir, thanks, please solve our problem

**Reply by dtrdic** [reply, marked as answer]
Please follow this steps;
1. Generate an SSH key pair with this command (check [here](https://docs.codemagic.io/knowledge-git/generating-an-ssh-key/) for reference)
`ssh-keygen -t ed25519 -f ~/Desktop/codemagic_ssh_key -q -N "" `
2. Copy the public key to your git repo 
3. Base64 your private key and copy it using this command (check [this](https://docs.codemagic.io/yaml-basic-configuration/configuring-environment-variables/#storing-binary-files) for reference)
`cat codemagic_ssh_key | base64 | pbcopy`
4. Your post request should look something like this
<img width="946" alt="Screenshot 2024-05-23 at 13 47 14" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/fad9f5c4-5e44-4f8f-9ad3-bde1400944bc">
